tests/contract/test_resource_client.py

```diff
@@ -302,6 +302,32 @@ def test_prune_properties():
     assert document == {"one": "two", "array": ["first"]}
 
 
+def test_prune_properties_for_all_sequence_members():
```

Comment:
Can we add an array of an object type and delete a property which is embedded in the object. This would give high confidence to the code

Reply:
Tests have been added.